app/assets/javascripts/discourse/app/components/deferred-render.gjs

```diff
@@ -0,0 +1,21 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import loadingSpinner from "discourse/helpers/loading-spinner";
+import runAfterFramePaint from "discourse/lib/after-frame-paint";
+
+export default class DeferredRender extends Component {
+  @tracked shouldRender = false;
+
+  constructor() {
+    super(...arguments);
+    runAfterFramePaint(() => (this.shouldRender = true));
+  }
+
+  <template>
+    {{#if this.shouldRender}}
+      {{yield}}
+    {{else}}
+      {{loadingSpinner}}
```

Comment:
so it tries to display a loading spinner for the duration of one frame paint? wouldn't it be better to display nothing at all?

Reply:
Yeah fair 👍